### PR TITLE
CA-398896: Add vlantest config to enable/disable vlantest

### PR DESCRIPTION
**Pull request by stephenchengCloud**

### CODE REVIEW

**Comment by DeliZhangX** [comment]
In my opinion:
- If want to remove it permanently (means never to run in certifying), we should delete code directly.
- If want to remain it as optional/internal test, we should add one option e.g. --vlan=true/false and false it's default; Also in xenrt test we can specify --vlan=true
- If specified in 'run_classes' option, the test should be able to execute.

**Comment by stephenchengCloud** [comment]
Code updated.
Tested that the vlan test is not generated without the "--vlantest" config.

**Review by catherineliuchina — COMMENTED** [review]
Did you run test result with acklogparser? Just want to confirm this change does not require change for acklogparser

File: autocertkit/network_tests.py [hunk 1]
```diff
@@ -464,6 +464,15 @@ class VLANTestClass(testbase.NetworkTestClass):
...
     num_ips_required = 4
 
+    def __init__(self, session, config):
+        super(VLANTestClass, self).__init__(session=session, config=config)
+        
+        # CA-398896: Skip vlan test from NetworkTest
+        # by removing the network base_tag 'NA'
+        if self.config['vlantest'] is False:
+            if 'NA' in self.tags:
+                self.tags.remove('NA')
```
Comment (DeliZhangX) [inline autocertkit/network_tests.py:474]: - Should not expose base class privacy `NA` here
- `TestClass` already has `extra_init` to extend than `__init__`

So could be 
```
class TestClass(object):


    def mask():
        if self.base_tag in self.tags:
            self.tags.remove(self.base_tag)
            
            
class VLANTestClass(testbase.NetworkTestClass):


    def extra_init(self):
        if not self.config['vlantest']:
            self.mask()
        
        super(VLANTestClass, self).extra_init()
```

For long term `tags` and test generator etc need to be refactored as they are not useful, while this is unrelated to your current change.

**Comment by stephenchengCloud** [comment]
> Did you run test result with acklogparser? Just want to confirm this change does not require change for acklogparser

This change doesn't affect acklogparser. The vlan test will not be generated.

File: autocertkit/network_tests.py (hunk 1 above)
Comment (stephenchengCloud) [reply autocertkit/network_tests.py]: code updated.

**Comment by stephenchengCloud** [comment]
> > Did you run test result with acklogparser? Just want to confirm this change does not require change for acklogparser
> 
> This change doesn't affect acklogparser. The vlan test will not be generated.

Parsed result with acklogparser, all went well.
Job id: 4150907
image

**Review by DeliZhangX — APPROVED** [review]

**Review by catherineliuchina — APPROVED** [review]
